web/docs/language/features.md

```diff
@@ -937,6 +943,115 @@ import AuthError from '@wasp/core/AuthError.js'
+
+This method requires that `externalAuthAssociationEntity` specified in `auth` [described here](features#externalauthassociationentity).
+#### Default settings
+- Configuration:
+  - By default, Wasp expects you to set two environment variables in order to use Google authentication:
+    - `GOOGLE_CLIENT_ID`
+    - `GOOGLE_CLIENT_SECRET`
+  - These can be obtained in your Google Cloud Console project dashboard. See [here](/docs/integrations/google#google-auth) for more.
+- Sign in:
+  - When a user signs in for the first time, Wasp will create a new User account and link it to their Google account for future logins. The `username` will default to a random dictionary phrase that does not exist in the database, like "nice-blue-horse-27160".
+    - Aside: If you would like to allow the user to select their own username, or some other sign up flow, you could add a boolean property to your User entity which indicates if the account setup is complete. You can then redirect them in your `onAuthSucceededRedirectTo` handler.
+- Here is a link to the default implementations: https://github.com/wasp-lang/wasp/blob/main/waspc/data/Generator/templates/server/src/routes/auth/passport/google/googleDefaults.js These can be overriden as explained below.
+
+#### Overrides
+If you require modifications to the above, you can add one or more of the following to your `auth.methods.google` dictionary:
+
+```js
+  auth: {
+    userEntity: User,
+    externalAuthAssociationEntity: SocialLogin,
+    methods: {
+      google: {
+        configFn: import { config } from "@ext/auth/google.js",
+        getUserFieldsFn: import { getUserFields } from "@ext/auth/google.js"
+      }
+    },
+    ...
+  }
+```
+
+- `configFn`: This function should return an object with the following shape:
+  ```js
+  export function config() {
+    // ...
+    return {
+      clientId, // look up from env or elsewhere,
+      clientSecret, // look up from env or elsewhere,
+      scope: ['profile'] // must include at least 'profile'
+    }
+  }
+  ```
+- `getUserFieldsFn`: This function should return the user fields to use when creating a new user upon their first Google login. The context contains a User entity for DB access, and the args are what the OAuth provider responds with. Here is how you could generate a username based on the Google display name. In your model, you could choose to add more attributes and set additional information.
+  ```js
+  import { generateAvailableUsername } from '@wasp/core/auth.js'
+
+  export async function getUserFields(_context, args) {
+    const username = await generateAvailableUsername(args.profile.displayName.split(' '), { separator: '.' })
+    return { username }
+  }
+  ```
+  - `generateAvailableUsername` takes an array of Strings and an optional separator and generates a string ending with a random number that is not yet in the database. For example, the above could produce something like "Jim.Smith.3984" for a Google user Jim Smith.
```

Comment:
Well we are really hepling them out here! In theory we could have just let them find such username on their own, right? Cool that we do it though!

Reply:
Yeah, I figured it was a decent utility function for them since we needed one very similar ourselves.